### PR TITLE
add hometask 3

### PR DESCRIPTION
**Pull request by lupolov-ivan**

### CODE REVIEW

File: src/main/java/ua/ithillel/odessa/simplerestapp/GreetingRepository.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            "carnivorous rebel",
+            "cursory intimidation",
+            "leased beggar",
+            "climactic handedness",
+            "obligated realist",
+            "disrupted docking",
+            "unkempt daemon"
+    );
+
+    public Map<String, String> getGreetingList() {
```
Comment (mfarsikov) [inline src/main/java/ua/ithillel/odessa/simplerestapp/GreetingRepository.java]: It is not recommended to use such suffixes as `List`, it is better just to use a plural name. 
Also, methods in repository usually are not 'getters', so good candidates for this method name are: `findGreetings`, `findAllGreetings`

File: src/main/java/ua/ithillel/odessa/simplerestapp/GreetingService.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        if(param.equals("random")) {
+            return getRandomStr();
+        }
+
+        return greetingRepository.getGreetingList().get(param);
+    }
+
+    public String getRandomStr() {
+        List<String> randomStrings = greetingRepository.getRandomWords();
+        Random random = new Random();
```
Comment (mfarsikov) [inline src/main/java/ua/ithillel/odessa/simplerestapp/GreetingService.java]: it is better not to create new Random for each method call.
So there are several options: 
* use `Math.random`
* move this variable to field
* create a `Random` bean and inject it!

File: src/test/java/ua/ithillel/odessa/simplerestapp/SimpleRestappApplicationTests.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                "obligated realist",
+                "disrupted docking",
+                "unkempt daemon")
+                .contains(gc.sayHello("random")));
+    }
+
+    @Test
+    public void uniformDistributionTest() {
+
+        double quantityTestStrings = 1000000; // количество строк в выборке
```
Comment (mfarsikov) [inline src/test/java/ua/ithillel/odessa/simplerestapp/SimpleRestappApplicationTests.java]: no cyrillic in source codes :(

File: src/test/java/ua/ithillel/odessa/simplerestapp/SimpleRestappApplicationTests.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        double delta = 0.001; // максимальное отклонение, полученное экспериментальным путем
+        double normalPercent = 0.1; // требуемый процент вхождения кажного слова чтобы распеределение был расномерным
+        List<String> testStrings = new ArrayList<>();
+
+        for (int i = 0; i < quantityTestStrings; i++) {
+            testStrings.add(gs.getRandomStr());
```
Comment (mfarsikov) [inline src/test/java/ua/ithillel/odessa/simplerestapp/SimpleRestappApplicationTests.java]: Can you rework it to Stream API?

File: src/test/java/ua/ithillel/odessa/simplerestapp/SimpleRestappApplicationTests.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    @Test
+    public void uniformDistributionTest() {
+
+        long quantityTestStrings = 1000000;
+        double delta = 0.001; // experimentally determined maximum deviation
+        double normalPercent = 0.1; // required percentage of occurrences of each word so that the distribution is uniform
+
+        List<String> testStrings = Stream.generate(gs::getRandomStr)
+                                         .limit(quantityTestStrings)
+                                         .collect(Collectors.toList());
```
Comment (mfarsikov) [inline src/test/java/ua/ithillel/odessa/simplerestapp/SimpleRestappApplicationTests.java]: this `collect` is redundant

Comment (lupolov-ivan) [reply src/test/java/ua/ithillel/odessa/simplerestapp/SimpleRestappApplicationTests.java]: Absolutely agree. I'll fix it.

**Pull request merged**